### DISCUSSION TITLE
How can I change the idekey from xDebug?

### DISCUSSION THREAD

**Discussion by lucyd-dev** [shyim/shopware-docker#107] [Q&A] [answered]
I'm using this solution under WSL 2. Everything works so far.
But when I want to use xDebug with VS Code, it doesn't work. There is a Connection when I run `telnet 0.0.0.0 9000` on Ubuntu, but I think it's because the idekey is set to `PHPSTORM`.

Is there any possibility to make it run?

**Comment by lucyd-dev** [comment]
Hm okay. To be sure, how can I debug a 500 Error on a product page with the following error?
`Object of class Shopware\Core\Checkout\Cart\Price\Struct\PriceCollection could not be converted to string`

**Comment by shyim** [comment, marked as answer]
I could set it up:

- Install: https://marketplace.visualstudio.com/items?itemName=felixfbecker.php-debug
- Enable XDebug for the specific SWDC project
  - Open file `~/.config/swdc/env` and add `VHOST_SW6_IMAGE="ghcr.io/shyim/shopware-docker/6/nginx:php80-xdebug3"`
  - `SW6` is my project folder name in upper case (VHOST_**SW6**_IMAGE)
- Restart swdc using swdc up
- Install chrome extension https://chrome.google.com/webstore/detail/xdebug-helper/eadndfjplgieldjbigjakmdgkmoaaaoc?hl=en
- Click on Debug in the extension
- Click in VSCode on Run -> Add Configuration

Add this config:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "port": 9000,
            "pathMappings": {
                "/var/www/html/sw6": "${workspaceFolder}"
            }
        }
    ]
}
```

You need to replace the `sw6` in the pathMappings with your project folder name. 

- Then start the XDebug Listing by opening the Run Tab and clicking on the Start icon
![image](https://user-images.githubusercontent.com/6224096/125138737-7c3c0780-e10f-11eb-9d17-eb1fd4f43a35.png)

![image](https://user-images.githubusercontent.com/6224096/125138764-8bbb5080-e10f-11eb-9c85-807a60602fc0.png)

**Reply by lucyd-dev** [reply]
I completely reinstalled everything and run swdc create-project sw6 to create a development project. I followed your steps (I use PHP7.4)  but I got this message from xdebug_info()
`[Step Debug] Could not connect to client host discovered through HTTP headers, connecting to configured address/port: localhost:9000. :-\|`

**Reply by shyim** [reply]
How is your VSCode running. With the WSL Extension and in the distro Ubuntu / Debian what ever? Or running on Windows host

**Reply by lucyd-dev** [reply]
It's running directly on  Ubuntu with the WSL extension.

**Reply by lucyd-dev** [reply]
Since I had to send my new laptop in for repair, I can now get back to you. I reinstalled everything and followed exactly your instructions (https://shyim.me/blog/shopware-development-environment-windows/) and your steps here, but it didn't work. After a while I tried again the option WSL_XDEBUG_TUNNEL=true and now it works perfectly. I don't know what my Problem last week was ^^'

Thanks!

**Reply by shyim** [reply]
Good to know 👍

**Reply by luminalpark** [reply]
@shyim do you think it's possible to reproduce the WSL_XDEBUG_TUNNEL workaround in dev-ops/docker/actions/start.sh in shopware/development?
